### ISSUE TITLE
Unhelpful error message on missing braces in an interface implementation

### ISSUE THREAD

**Issue by jiribenes** [effekt-lang/effekt#369]
When trying to write an instance of an interface:
```scala
interface Eff {
  def op(): Int
}

def e: Eff = new Eff {
  def op() = println("42"); 42
}
```

shows an error: "Operation Eff is not part of interface Eff."

Here's a screenshot:
<img width="518" alt="Screenshot 2024-01-23 at 17 19 27" src="https://github.com/effekt-lang/effekt/assets/11269173/dac5de6a-cb29-4f2a-8a42-15021e9247f7">

One way to work around this is to add braces into the method body:
```scala
interface Eff {
  def op(): Int
}

def e: Eff = new Eff {
  def op() = { println("42"); 42 }
}
```

**Comment by b-studios** [comment]
TBH it took me quite a while to understand this error...

The parser parses it as 
```
def e: Eff = new Eff { () =>
  def op() = println("42")
  42
}
```
which is our syntax for singleton operations.

**Comment by b-studios** [comment]
@jiribenes what would you expect as an error message? We could also forbid parsing it as single operation in case it starts with `def`

**Comment by b-studios** [comment]
I wouldn't know an elegant way to prevent it though. 

https://github.com/effekt-lang/effekt/blob/49f2166f85c228b6169d1e7d0c1f508081c5551a/effekt/shared/src/main/scala/effekt/Parser.scala#L426-L437

The parser for implementation reuses `functionArg` which admits the program you wrote.

**Comment by jiribenes** [comment]
After #495, the error looks something like this:
![image](https://github.com/effekt-lang/effekt/assets/11269173/568e54be-896c-49cb-a049-eea294eb2ffd)

which means I can somewhat improve it (or at least underline the relevant part) by consuming tokens after `;` until I find `}` or `def` or EOF:
<img width="1062" alt="Screenshot 2024-06-22 at 16 36 28" src="https://github.com/effekt-lang/effekt/assets/11269173/19f48a05-b56c-4613-810c-485607099ac3">

**Comment by jiribenes** [comment]
I resurrected the fix from the comment above in #625.
